modules/component/src/core/cd-aware/cd-aware_creator.ts

```diff
@@ -43,40 +37,69 @@ export function setUpWork(cfg: WorkConfig): () => void {
  * Also custom behaviour is something you need to implement in the extending class
  */
 export function createCdAware<U>(cfg: {
-  work: () => void;
-  resetContextObserver: NextObserver<unknown>;
-  configurableBehaviour: (
-    o: Observable<Observable<U | null | undefined>>
-  ) => Observable<Observable<U | null | undefined>>;
-  updateViewContextObserver: PartialObserver<U | null | undefined>;
+  strategies: StrategySelection<U>;
+  resetContextObserver: NextObserver<undefined>;
+  updateViewContextObserver: PartialObserver<U> & NextObserver<U>;
 }): CdAware<U | undefined | null> {
-  const observablesSubject = new Subject<
-    Observable<U> | Promise<U> | null | undefined
-  >();
-  const observables$: Observable<
-    U | undefined | null
-  > = observablesSubject.pipe(
-    distinctUntilChanged(),
-    // Try to convert it to values, throw if not possible
-    map(toObservableValue),
-    tap((v: any) => {
-      cfg.resetContextObserver.next(v);
-      cfg.work();
-    }),
-    map(value$ =>
-      value$.pipe(distinctUntilChanged(), tap(cfg.updateViewContextObserver))
-    ),
-    cfg.configurableBehaviour,
-    switchAll(),
-    tap(() => cfg.work())
+  const strategyNameSubject = new BehaviorSubject<string>(
+    DEFAULT_STRATEGY_NAME
+  );
+  const strategy$: Observable<CdStrategy<U>> = strategyNameSubject.pipe(
```

Comment:
I implemented it based on the fact that it is technically possible to change it as these are input bindings. Therefore, some users will do it.

This situation should not throw errors nor work unexpectedly. (even if it might happen rarely)

This was the reason for implementing it in that way. If it can change I have to handle it correctly.

Reply:
I asked some peoples that experimented with the different rendering approaches. 
AFAICT they had the use-case to chance the rendering based on some used dynamic components. I will analyze that one and get back with infos and solutions.

Reply:
It helps a lot in testing.

But I kick it out!

Reply:
Fixed